crates/rustc_codegen_spirv/src/builder/spirv_asm.rs

```diff
@@ -1357,6 +1358,61 @@ impl<'cx, 'tcx> Builder<'cx, 'tcx> {
+                Err(()) => self.err(format!("unknown QuantizationModes {word}")),
+            },
+            (OperandKind::FPOperationMode, Some(word)) => match word.parse() {
+                Ok(x) => inst.operands.push(dr::Operand::FPOperationMode(x)),
+                Err(()) => self.err(format!("unknown FPOperationMode {word}")),
+            },
+            (OperandKind::OverflowModes, Some(word)) => match word.parse() {
+                Ok(x) => inst.operands.push(dr::Operand::OverflowModes(x)),
+                Err(()) => self.err(format!("unknown OverflowModes {word}")),
+            },
+            (OperandKind::PackedVectorFormat, Some(word)) => match word.parse() {
+                Ok(x) => inst.operands.push(dr::Operand::PackedVectorFormat(x)),
+                Err(()) => self.err(format!("unknown PackedVectorFormat {word}")),
+            },
+            (OperandKind::HostAccessQualifier, Some(word)) => match word.parse() {
+                Ok(x) => inst.operands.push(dr::Operand::HostAccessQualifier(x)),
+                Err(()) => self.err(format!("unknown HostAccessQualifier {word}")),
+            },
+            (OperandKind::CooperativeMatrixOperands, Some(word)) => {
+                match parse_bitflags_operand(COOPERATIVE_MATRIX_OPERANDS, word) {
+                    Some(x) => inst
+                        .operands
+                        .push(dr::Operand::CooperativeMatrixOperands(x)),
+                    None => self.err(format!("Unknown CooperativeMatrixOperands {word}")),
+                }
+            }
+            (OperandKind::CooperativeMatrixLayout, Some(word)) => match word.parse() {
+                Ok(x) => inst.operands.push(dr::Operand::CooperativeMatrixLayout(x)),
+                Err(()) => self.err(format!("unknown CooperativeMatrixLayout {word}")),
+            },
+            (OperandKind::CooperativeMatrixUse, Some(word)) => match word.parse() {
+                Ok(x) => inst.operands.push(dr::Operand::CooperativeMatrixUse(x)),
+                Err(()) => self.err(format!("unknown CooperativeMatrixUse {word}")),
+            },
+            (OperandKind::InitializationModeQualifier, Some(word)) => match word.parse() {
+                Ok(x) => inst
+                    .operands
+                    .push(dr::Operand::InitializationModeQualifier(x)),
+                Err(()) => self.err(format!("unknown InitializationModeQualifier {word}")),
+            },
+            (OperandKind::LoadCacheControl, Some(word)) => match word.parse() {
+                Ok(x) => inst.operands.push(dr::Operand::LoadCacheControl(x)),
+                Err(()) => self.err(format!("unknown LoadCacheControl {word}")),
+            },
+            (OperandKind::StoreCacheControl, Some(word)) => match word.parse() {
+                Ok(x) => inst.operands.push(dr::Operand::StoreCacheControl(x)),
+                Err(()) => self.err(format!("unknown StoreCacheControl {word}")),
+            },
```

Comment:
Wow I didn't realize this was so manual. Longer-term might be worth switching to SPIR-T's view of `OperandKind`, which handles immediate operands in a much more dynamic way, and contains a lot of the same information that `.parse()` uses in these `match` arms (but then I'm not sure you could create the typed `rspirv` `Operand` value out of that).